src/parallel.jl

```diff
@@ -0,0 +1,143 @@
+using Base.Threads
+using Base.Threads: @spawn, fetch
+
+
+struct Partition
+    nt::Int
+    n::Int
+    k::Int
+end
+Partition(nt,n) = Partition(nt, n, div(n, nt))
+Base.length(pt::Partition) = pt.nt
+(pt::Partition)(i) = div((i-1), pt.k) + 1, i - pt.k*div((i-1), pt.k)
+(pt::Partition)(q1, q2) = (q1-1)*pt.k + q2 
+
+each(pt::Partition) = 1:pt.nt
+
+function parallel_innermost!(partition, G, G1, G2, ∇ϕ, i, t, x, θ, t′, Q, c, b, t_old,
+    F::Union{ZigZag,FactBoomerang}, (factor, adapt), args...)
+    t, x, θ = smove_forward!(G, i, t, x, θ, t′, F)
+    ∇ϕi = ∇ϕ(x, i, args...)
+    l, lb = sλ(∇ϕi, i, x, θ, F), sλ̄(b[i], t[i] - t_old[i])
+    num = 1
+    if rand()*lb < l
+        if l >= lb
+            !adapt && error("Tuning parameter `c` too small.")
+            adapt!(c, i, factor)
+        end
+        t, x, θ = smove_forward!(G2, i, t, x, θ, t′, F)
```

Comment:
This is save if `i` is inner! (These are those for which I need the position to compute a good bound, and they should be in the same partition.)